src/System.IO.FileSystem/src/System/IO/Directory.cs

```diff
@@ -579,6 +579,11 @@ public static void Delete(String path, bool recursive)
             String fullPath = Path.GetFullPath(path);
             FileSystem.Current.RemoveDirectory(fullPath, recursive);
         }
+
+        public static string[] GetLogicalDrives()
```

Comment:
> I'd also like to see the existing less-than-perfect bits drop off of FileSystem long-term

That's a fine goal.  But we're not currently exposing this abstraction, and I expect that if/when we do, it would need to change in some notable ways.  We should not then harm the current implementation by having yet another way of providing a PAL implementation that conflicts.

> I'll also note that it feels "icky".

I'm not seeing the issue.  We have an abstract GetLogicalDrives member in FileSystem.  In UnixFileSystem we override that method to do ```return Interop.Sys.GetAllMountPoints().ToArray();```.  In Win32FileSystem we override that method to do ```return DriveInfoInternal.GetLogicalDrives();```. In WinRTFileSystem we override that method to do ```throw new PlatformNotSupportedException();``` (unless there is a way to implement... I'm basing that purely on your implementation).  That DriveInfoInternal class exists because that method is also consumed into System.IO.FileSystem.DriveInfo.dll, but we only need the  one file that's part of it, not the other one you defined, as that's replaced by the abstract method override.  How is that "icky"?  Seems quite clean to me.